inference-engine/src/cldnn_engine/cldnn_engine.cpp

```diff
@@ -106,6 +107,12 @@ InferenceEngine::ICNNNetwork::Ptr clDNNEngine::CloneNetwork(const InferenceEngin
 
         manager.set_callback(transformations_callback);
         manager.run_passes(nGraphFunc);
+
+        // Apply all transformations to TensorIterator body
+        ngraph::pass::Manager ti_manager;
+        ti_manager.register_pass<ngraph::pass::ApplyTransformationsToTIBody>(manager);
```

Comment:
Yes, you are right, now an instance of Manager is required. It might be much better to send a vector of Passes to this transformation and add a getter to the Manager class. What do you think? 

>>  Is there a way to "forward" each transformation to TI and other operations that applicable by adding some virtual forwarding method to Node class and let sub-classes to override it?
Not investigated.

Reply:
I have some solutions in my head how we can apply transformations in TI body:

1. Update GraphRewrite to capture TI nodes and recursively execute on its body converted to Function. But in this case we can not automatically update all user transformations based on FunctionPass, so it requires changes in every FunctionPass transformation. Moreover it is not scalable if in some day we decide to add another operation that contains Function inside then we need to update this passes again.
2. Update pass Manager to work with Function inside operations. In this case we don't have to updated each transformation and pass Manager will do all dirty work. But I don't want ngrpah structures know about operations and their features. Also within approach I don't like that you can not execute transformation without pass Manager if you want to change ops with sub-graphs.
3. Have a dedicated transformation that applies given transformations to all sub-graphs in ops. This transformation can take Manager as input (because it's handy container for all type of transformations). In this case we don't tied up ngraph structures with specific operations and it requires small changes. But we need to execute this transformation everywhere where we execute regular transformations and that is what we pay.

So in this PR I choose third approach as it requires minimal changes. But anyway its not the end of the story and we need to figure out how we can improve it.